pages/flight_list.js

```diff
@@ -12,9 +17,69 @@ export default function FlightList() {
     router.push("/add-flight");
   };
 
-  const [flightsOfInterest] = useLocalStorageState("flightsInfo", {
-    defaultValue: flights,
-  });
+  const [flightsOfInterest, setFlightsOfInterest] = useLocalStorageState(
+    "flightsInfo",
+    {
+      defaultValue: flights,
+    }
+  );
+
+  const [isEditModalOpen, setIsEditModalOpen] = useState(false);
+  const [selectedFlight, setSelectedFlight] = useState(null);
+
+  const handleEditClick = (flight) => {
+    // Access the flight data from local storage
+    const storedFlights = flightsOfInterest;
+
+    // Find the specific flight that corresponds to the clicked IconButton
+    const selectedFlight = storedFlights.find(
+      (storedFlight) =>
+        storedFlight.flight_iata === flight.flight_iata &&
+        storedFlight.scheduled_date === flight.scheduled_date
+    );
+
+    setIsEditModalOpen(true);
+    setSelectedFlight(selectedFlight);
+  };
+
+  const handleEditModalSave = (editedFlightData) => {
+    // Update the flight data in main state (flightsOfInterest)
+    const updatedFlights = flightsOfInterest.map((flight) => {
+      if (
+        flight.flight_iata === editedFlightData.flight_iata &&
+        flight.scheduled_date === editedFlightData.scheduled_date
+      ) {
```

Comment:
The additional info is accepted here and passed along to check for the correct flight entry.
```suggestion
  const handleEditModalSave = (iata, date, editedFlightData) => {
    // Update the flight data in main state (flightsOfInterest)
    const updatedFlights = flightsOfInterest.map((flight) => {
      if (
        flight.flight_iata === iata &&
        flight.scheduled_date === date
```